idb/postgres/internal/writer/writer.go

```diff
@@ -269,8 +278,9 @@ func writeAppResource(round basics.Round, resource *ledgercore.AppResourceRecord
 	if resource.Params.Params != nil && resource.Params.Params.ApprovalProgram != nil {
 		b64 := base64.StdEncoding
 		var appHash = b64.EncodeToString([]byte(resource.Params.Params.ApprovalProgram))
+		SigmaDAOApp, err := readSigmaDAOApp()
```

Comment:
Why re-reading it on every function call?

Reply:
I have added support to only read from file when app hash has changed.